src/main/java/org/tc/services/course/CourseServiceImpl.java

```diff
@@ -119,7 +150,51 @@ public boolean isProposal(Course course) {
 
     @Override
     public boolean isRejected(Course course) {
-        return course.getState().equals(StateEnum.PROPOSAL);
+        return course.getState().equals(StateEnum.REJECTED);
+    }
+
+    @Override
+    public boolean isNew(Course course) {
+        return course.getState().equals(StateEnum.NEW);
+    }
+
+    @Override
+    public boolean isOpen(Course course) {
+        return course.getState().equals(StateEnum.OPEN);
+    }
+
+    @Override
+    public boolean isReady(Course course) {
```

Comment:
Such methods makes sense to move to model

course.isReady()
cource.isInProgress()

better.